text/0124-extrinsic-version-5.md

```diff
@@ -0,0 +1,156 @@
+
+## Stakeholders
+
+- Runtime users
+- Runtime devs
+- Wallet devs
+
+## Explanation
+
+### Changes to extrinsic authorization
+
+The introduction of `General` transactions allows the authorization of any and all origins through
+extensions. This means that, with the appropriate extension, `General` transactions can replicate
+the same behavior present-day v4 `Signed` transactions. Specifically for Polkadot chains, an example
+implementation for such an extension is
+[`VerifySignature`](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/verify-signature),
+introduced in the Transaction Extension
+[PR3685](https://github.com/paritytech/polkadot-sdk/pull/3685). Other extensions can be inserted
+into the extension pipeline to authorize different custom origins. Therefore, a `Signed` extrinsic
+variant is redundant to a `General` one strictly in terms of user functionality and could eventually
+be deprecated and removed.
+
+### Encoding format for version 5
+
+As with version 4, the encoded extrinsic v5 is a SCALE encoded vector of bytes (`u8`), therefore
+starting with the encoded length of the following bytes in compact format. The leading byte after
+the length determines the version and type of extrinsic, as specified by
+[RFC84](https://github.com/polkadot-fellows/RFCs/blob/main/text/0084-general-transaction-extrinsic-format.md).
+For reasons mentioned above, this RFC removes the `Signed` variant for v5 extrinsics.
+
+NOTE: For `Bare` extrinsics, the following bytes will just be the encoded call and nothing else.
+
+For `General` transactions, as stated in
+[RFC99](https://github.com/polkadot-fellows/RFCs/blob/main/text/0099-transaction-extension-version.md),
+an extension version byte must be added in the next extrinsic version. This byte should allow
+runtimes to expose more than one set of extensions which can be used for a transaction. As far as
+the v5 extrinsic encoding is concerned, this extension byte should be encoded immediately after the
+leading encoding byte. The extension version byte should be included in payloads to be signed by all
+extensions configured by runtime devs to ensure a user's extension version choice cannot be altered
+by third parties.
+
+After the extension version byte, the extensions will be encoded next, followed by the call itself.
+
+A quick visualization of the encoding:
+
+- `Bare` extrinsics: `(extrinsic_encoded_len, 0b0000_0101, call)`
+- `General` transactions: `(extrinsic_encoded_len, , 0b0100_0101, extension_version_byte, extension, call)`
```

Comment:
The extension itself is usually a tuple of multiple extensions, generally referred to as the extension pipeline. Technically it's only one extension, the `TxExtension` commonly defined in runtimes, but that is always a tuple of extensions like `CheckNonce`, `CheckWeight`, `ChargeTransactionPayment` etc., so it would be only one extension version, as it is the version of the tuple, but there are multiple extension**s** in the pipeline.